src/network/network_gui.cpp

```diff
@@ -2371,6 +2372,16 @@ struct NetworkClientListWindow : Window {
 
 				break;
 			}
+
+			case WID_CL_CLIENT_NUMBER: {
+				/* Number of clients */
+				int text_y_offset = std::max(0, ((int)this->line_height - (int)FONT_HEIGHT_NORMAL) / 2);
```

Comment:
`this->line_height` is set to the maximum of some height and `FONT_HEIGHT_NORMAL`, so that value will always be a positive number. So no need for `int` or `std::max` here.
Having said that, take a look at `SetDataTip` and how it's used with `NWidget(WWT_TEXTBTN, ...`. I think by adding a `SetDataTip(STR_NETWORK_CLIENT_LIST_CLIENT_NUMBER, <some to be defined string for the tooltip>)` to line 1634, and putting the `SetDParam` lines below in case of `SetStringParameters` around line 2040, you do not need all this custom calculation and drawing.

Reply:
Thanks for details, it was a beating my head against a stone wall 😛